### PR TITLE
Patch kong insomnia

### PR DESCRIPTION
**Pull request by MichaelSp**
The versioning scheme of insomnia and inso has changed from 2023.5.8 to 8.x.x
see https://insomnia.rest/changelog#8.0.0

---
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/i/insomnia.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@
   homepage "https://insomnia.rest/"
 
   livecheck do
-    regex(/core@v?(\d+(?:\.\d+)+)(?!.)/i)
+    regex(/core@v?(\d{1,3}(?:\.\d+)+)(?!.)/i)
```
Comment (bevanjkay) [inline Casks/i/insomnia.rb:12]: An alternative to consider could be to explicitly negate the `20##` versions.
```suggestion
    regex(/core@v?((?!20\d+)\d+?(?:\.\d+)+)(?!.)/i)
```

Comment (samford) [reply Casks/i/insomnia.rb:12]: I'm good with avoiding the date-based versions by restricting the number of digits (that's how I typically handle this particular situation). I'll push some changes in a moment to add explanatory comments and tweak the regexes a little.

**Review by X-oss-byte — COMMENTED** [review]
Cask

**Comment by miccal** [comment]
Thank you @MichaelSp and @samford.

**Review by miccal — APPROVED** [review]

**Pull request merged**